app/javascript/components/common/ComboButton.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useContext } from 'react'
+import ReactDOM from 'react-dom'
+import { usePanel } from '../../hooks/use-panel'
+import { GraphicalIcon } from './GraphicalIcon'
+
+const ComboButtonContext = React.createContext({
+  open: false,
+  setPanelElement: (
+    value: React.SetStateAction<HTMLDivElement | null>
+  ): void => {},
+  styles: {},
+  attributes: {},
+})
+
+export const ComboButton = ({
+  className = '',
+  children,
+}: React.PropsWithChildren<{ className?: string }>): JSX.Element => {
+  const {
+    open,
+    setOpen,
+    setButtonElement,
+    setPanelElement,
+    styles,
+    attributes,
+  } = usePanel({
+    placement: 'bottom-end',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 14],
+        },
+      },
+    ],
+  })
+
+  const classNames = ['c-combo-button', className]
+
+  return (
+    <ComboButtonContext.Provider
+      value={{
+        setPanelElement: setPanelElement,
+        open: open,
+        styles: styles,
+        attributes: attributes,
+      }}
+    >
+      <div className={classNames.join(' ')}>
+        {children}
+        <div className="--dropdown-segment">
+          <button
```

Comment:
The button should be the `--dropdown-segment`. This extra div shouldn't be here.